### PR TITLE
Generate managers and projects for Azure Libraries for .NET

### PR DESCRIPTION
**Pull request by RikkiGibson**
- Adds templates for Manager, .csproj, and AssemblyInfo
- Assumes output-folder doesn't end with a Generated/ component so these "project structure" outputs can be written at the top level
- Migrates test fixtures to new path

### CODE REVIEW

**Comment by olydis** [comment]
# 🤖 AutoRest automatic publish job 🤖
~~~ Haskell
> will publish once PR gets merged
~~~

**Comment by RikkiGibson** [comment]
Looks like I need to merge upstream/master and regenerate fluent fixtures.

**Comment by RikkiGibson** [comment]
@olydis any concerns related to this?

File: src/azure/Templates/AzureCsprojTemplate.cshtml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    <PropertyGroup>
+        <PackageTargetFallback Condition=" '$(TargetFramework)' == 'netstandard1.3' ">$(PackageTargetFallback);dnxcore50</PackageTargetFallback>
+        <NetStandardImplicitPackageVersion Condition=" '$(TargetFramework)' == 'netstandard1.3' ">1.6.1</NetStandardImplicitPackageVersion>
+        <TargetFrameworks>net452;netstandard1.3</TargetFrameworks>
+    </PropertyGroup>
+
+    @* The "Version=" needs to be inside a C# expression in order to prevent the compiled template from calling WriteAttributeValue which causes a build error. *@
+    <ItemGroup>
+        <PackageReference Include="Microsoft.Rest.ClientRuntime.Azure" @("Version=\"" + @Model.ClientRuntimeVersion + "\"") />
+        <PackageReference Include="Microsoft.Azure.Management.ResourceManager.Fluent" @("Version=\"" + @Model.ClientPackageVersion + "\"") />
```
Comment (olydis) [inline src/azure/Templates/AzureCsprojTemplate.cshtml]: this template contains fluent-specific stuff so it should go into the `azurefluent` folder ;)

File: src/azure/Templates/AzureServiceManagerTemplate.cshtml [hunk 2]
```diff
@@ -0,0 +1,115 @@
+﻿@using System
```
Comment (olydis) [inline src/azure/Templates/AzureServiceManagerTemplate.cshtml]: same here, doesn't belong into `azure` folder

File: src/azure/Model/CodeModelCsa.cs [hunk 3]
```diff
@@ -32,5 +36,56 @@ public override IEnumerable<string> Usings
...
+
+                var method = Methods[0];
+                var match = Regex.Match(input: method.Url, pattern: @"/providers/microsoft\.(\w+)/", options: RegexOptions.IgnoreCase);
+                var serviceName = match.Groups[1].Value.ToPascalCase();
+                return serviceName;
+            }
+        }
+
+        [JsonIgnore]
+        public string ClientRuntimeVersion
```
Comment (olydis) [inline src/azure/Model/CodeModelCsa.cs]: consider just doing
```
Settings.Instance.Host?.GetValue<string>("runtime-version").Result ?? "{VERSION}"
```
less control flowy that way 😉

**Pull request merged**